Investigate flaky test EventClientITest#shouldListEventsAsyncWithQueryOptionsAndCallback

Here are a few errors from recent runs in CI. I only see these failures occasionally in CI.

```
[INFO] Running org.kiwiproject.consul.EventClientITest
Error:  Tests run: 9, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 2.317 s <<< FAILURE! -- in org.kiwiproject.consul.EventClientITest
Error:  org.kiwiproject.consul.EventClientITest.shouldListEventsAsyncWithCallback -- Time elapsed: 2.025 s <<< ERROR!
org.awaitility.core.ConditionTimeoutException: 
Assertion condition defined as a Lambda expression in org.kiwiproject.consul.EventClientITest 
Expecting all elements of:
  ["f006b1c2-55c8-931c-1e7d-0448cb663dd8",
    "2b89be8d-2a1a-e025-893c-05ba31c7b474",
    "2acd2dc9-c7e3-20c5-6368-3eed622bcba3",
    "1b884d93-c4b2-abe6-3f99-61627eaf3849",
    "8fcb6cf2-a119-15e7-e1e5-fff1791d5f86",
    "31678416-5767-e3c0-5d5a-b48a039614be",
    "abe15386-078f-fd43-1491-efc2f7050987",
    "f77398e0-dee0-a6db-d810-d05dbeb15d5f",
    "d371d656-85ff-92e0-e40a-e8485272f06c",
    "5f6a1cd0-32a8-fdbf-85c0-af2f3ba6852c"]
to match given predicate but this element did not:
  "5f6a1cd0-32a8-fdbf-85c0-af2f3ba6852c" within 2 seconds.
	at org.awaitility.core.ConditionAwaiter.await(ConditionAwaiter.java:167)
	at org.awaitility.core.AssertionCondition.await(AssertionCondition.java:119)
	at org.awaitility.core.AssertionCondition.await(AssertionCondition.java:31)
	at org.awaitility.core.ConditionFactory.until(ConditionFactory.java:1160)
	at org.awaitility.core.ConditionFactory.untilAsserted(ConditionFactory.java:790)
	at org.kiwiproject.consul.EventClientITest.shouldListEventsAsyncWithCallback(EventClientITest.java:121)
Caused by: java.lang.AssertionError: 
Expecting all elements of:
  ["f006b1c2-55c8-931c-1e7d-0448cb663dd8",
    "2b89be8d-2a1a-e025-893c-05ba31c7b474",
    "2acd2dc9-c7e3-20c5-6368-3eed622bcba3",
    "1b884d93-c4b2-abe6-3f99-61627eaf3849",
    "8fcb6cf2-a119-15e7-e1e5-fff1791d5f86",
    "31678416-5767-e3c0-5d5a-b48a039614be",
    "abe15386-078f-fd43-1491-efc2f7050987",
    "f77398e0-dee0-a6db-d810-d05dbeb15d5f",
    "d371d656-85ff-92e0-e40a-e8485272f06c",
    "5f6a1cd0-32a8-fdbf-85c0-af2f3ba6852c"]
to match given predicate but this element did not:
  "5f6a1cd0-32a8-fdbf-85c0-af2f3ba6852c"
	at org.kiwiproject.consul.EventClientITest.lambda$shouldListEventsAsyncWithCallback$6(EventClientITest.java:123)
```

Similarly:

```
[INFO] Running org.kiwiproject.consul.EventClientITest
Error:  Tests run: 9, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 2.342 s <<< FAILURE! -- in org.kiwiproject.consul.EventClientITest
Error:  org.kiwiproject.consul.EventClientITest.shouldListEventsAsyncWithQueryOptionsAndCallback -- Time elapsed: 2.036 s <<< ERROR!
org.awaitility.core.ConditionTimeoutException: 
Assertion condition defined as a Lambda expression in org.kiwiproject.consul.EventClientITest 
Expecting all elements of:
  ["5eaa9250-6aa1-5f01-b19b-db0265248f8c",
    "741f8e0c-adef-8e5d-085b-e14c242fe0d1",
    "68454f12-82f8-f4e9-9b5b-89c8ab27c0fe",
    "1f507f70-74f3-8f36-8a48-c8ae34405424",
    "bc015bce-8123-2502-94cf-0b67d8b41077",
    "9ba96a55-e53f-1fbf-dab7-49fc1dd011e2",
    "9dd15b66-c7c4-9ae2-aa52-50974cd63c80",
    "1c7092dc-1904-9195-37a9-96bc602951b3",
    "8434d145-522d-cb83-dd42-30220936345f",
    "79c51b49-e0d7-5704-20f3-f7d1c212efae"]
to match given predicate but this element did not:
  "79c51b49-e0d7-5704-20f3-f7d1c212efae" within 2 seconds.
	at org.awaitility.core.ConditionAwaiter.await(ConditionAwaiter.java:167)
	at org.awaitility.core.AssertionCondition.await(AssertionCondition.java:119)
	at org.awaitility.core.AssertionCondition.await(AssertionCondition.java:31)
	at org.awaitility.core.ConditionFactory.until(ConditionFactory.java:1160)
	at org.awaitility.core.ConditionFactory.untilAsserted(ConditionFactory.java:790)
	at org.kiwiproject.consul.EventClientITest.shouldListEventsAsyncWithQueryOptionsAndCallback(EventClientITest.java:139)
Caused by: java.lang.AssertionError: 
Expecting all elements of:
  ["5eaa9250-6aa1-5f01-b19b-db0265248f8c",
    "741f8e0c-adef-8e5d-085b-e14c242fe0d1",
    "68454f12-82f8-f4e9-9b5b-89c8ab27c0fe",
    "1f507f70-74f3-8f36-8a48-c8ae34405424",
    "bc015bce-8123-2502-94cf-0b67d8b41077",
    "9ba96a55-e53f-1fbf-dab7-49fc1dd011e2",
    "9dd15b66-c7c4-9ae2-aa52-50974cd63c80",
    "1c7092dc-1904-9195-37a9-96bc602951b3",
    "8434d145-522d-cb83-dd42-30220936345f",
    "79c51b49-e0d7-5704-20f3-f7d1c212efae"]
to match given predicate but this element did not:
  "79c51b49-e0d7-5704-20f3-f7d1c212efae"
	at org.kiwiproject.consul.EventClientITest.lambda$shouldListEventsAsyncWithQueryOptionsAndCallback$8(EventClientITest.java:141)
```

--- a/src/itest/java/org/kiwiproject/consul/EventClientITest.java
+++ b/src/itest/java/org/kiwiproject/consul/EventClientITest.java
@@ -115,6 +115,12 @@ class EventClientITest extends BaseIntegrationTest {
                 .describedAs("Precondition: Event IDs should not be empty")
                 .isNotEmpty();
 
+        // Confirm all events are visible synchronously before exercising the async path.
+        // The async callback fires once; if Consul hasn't indexed all events yet that
+        // response is frozen with an incomplete set and the assertion can never recover.
+        awaitAtMost2s().untilAsserted(() ->
+                assertThat(getEventIds(eventClient.listEvents())).containsAll(eventIds));
+
         var callback = new TestEventResponseCallback();
         eventClient.listEvents(callback);
 
@@ -132,6 +138,12 @@ class EventClientITest extends BaseIntegrationTest {
         assertThat(eventIds)
                 .describedAs("Precondition: Event IDs should not be empty")
                 .isNotEmpty();
+
+        // Confirm all events are visible synchronously before exercising the async path.
+        // The async callback fires once; if Consul hasn't indexed all events yet that
+        // response is frozen with an incomplete set and the assertion can never recover.
+        awaitAtMost2s().untilAsserted(() ->
+                assertThat(getEventIds(eventClient.listEvents())).containsAll(eventIds));
 
         var callback = new TestEventResponseCallback();
         eventClient.listEvents(Options.BLANK_QUERY_OPTIONS, callback);

